### PR TITLE
Add downcase transformer

### PR DESCRIPTION
**Pull request by spheric**
[MableEtl gem add downcase to gem and update Mable app](https://bettercaring.atlassian.net/browse/ES-16721)

### CODE REVIEW

File: lib/transformers/downcase_transformer.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            row_data = row[column]
+
+            next unless row_data.is_a?(String)
+
+            row[column] = row_data.downcase
+          end
+
+          row
+        end
+        # binding.pry
```
Comment (spheric) [inline lib/transformers/downcase_transformer.rb:34]: binding.pry comment

File: lib/mable_etl/control.rb [hunk 2]
```diff
@@ -47,5 +60,15 @@ def transform
     def load
       @load ||= MableEtl::Loaders::LoaderFactory.for(params).load
     end
+
+    def log_result(result)
```
Comment (spheric) [inline lib/mable_etl/control.rb:64]: Needs test cases for 

* when extracting logger receives info with result.message
* when extracting logger receives error with result.message
* when transforming logger receives info with result.message
* when transforming logger receives error with result.message

**Review by spheric — COMMENTED** [review]
looks good, some more test cases required

File: spec/mable_etl/control_spec.rb [hunk 3]
```diff
@@ -2,29 +2,56 @@
...
     end
+
+    context 'extractor' do
+      context 'when it is successful' do
+        it 'should receive info' do
+          expect(dummy_logger).to receive(:info)
+          control.process
+        end
+
+        context 'when it is successful' do
```
Comment (spheric) [inline spec/mable_etl/control_spec.rb:43]: unsuccessful

File: spec/mable_etl/control_spec.rb [hunk 4]
```diff
@@ -2,29 +2,56 @@
...
-      end
       it 'is returns a loader object' do
         expect(control.process).to eq(loader_result)
       end
     end
+
+    context 'extractor' do
+      context 'when it is successful' do
+        it 'should receive info' do
+          expect(dummy_logger).to receive(:info)
```
Comment (spheric) [inline spec/mable_etl/control_spec.rb:39]: better to be explicit here `expect(dummy_logger).to have_received(:info).with(message)`

File: spec/mable_etl/control_spec.rb [hunk 5]
```diff
@@ -2,29 +2,56 @@
...
+        end
+
+        context 'when it is successful' do
+          let(:extract) { instance_double(MableEtl::Extractors::ExtractorResult, success?: false, mable_etl_file_path: 'file_path', message: 'abc') }
+
+          before do
+            allow(MableEtl::Extractors::ExtractorResult).to receive(:new).and_return(extract)
+          end
+          it 'it returns a logger error' do
+            expect(dummy_logger).to receive(:error)
```
Comment (spheric) [inline spec/mable_etl/control_spec.rb:50]: better to be explicit here `expect(dummy_logger).to have_received(:error).with('abc')`

File: spec/mable_etl/control_spec.rb [hunk 6]
```diff
@@ -2,29 +2,56 @@
 
 require 'spec_helper'
 require 'mable_etl/control'
+require 'fixtures/dummy_logger'
 
 RSpec.describe MableEtl::Control do
```
Comment (spheric) [inline spec/mable_etl/control_spec.rb:7]: need specs for when transformer logs result as well

File: spec/mable_etl/control_spec.rb [hunk 7]
```diff
@@ -2,29 +2,82 @@
...
+          end
+
+          context 'when it is unsuccessful' do
+            let(:transform) { instance_double(MableEtl::Transformers::TransformerResult, success?: false, mable_etl_data: 'file_path', message: 'abc') }
+
+            before do
+              allow(MableEtl::Transformers::TransformerResult).to receive(:new).and_return(transform)
+            end
+            it 'it returns a logger error' do
+              # control.process
```
Comment (spheric) [inline spec/mable_etl/control_spec.rb:73]: hmm why didn't this work?

File: spec/mable_etl/control_spec.rb [hunk 8]
```diff
@@ -2,28 +2,81 @@
...
+        allow(MableEtl::Extractors::ExtractorResult).to receive(:new).and_return(extract)
+        allow(MableEtl::Transformers::TransformerResult).to receive(:new).and_return(transform)
       end
-      it 'is returns a loader object' do
-        expect(control.process).to eq(loader_result)
+
+      context 'when it is successful' do
+        it 'should receive info' do
+          dummy_logger = double
+          allow(dummy_logger).to receive(:info).and_return(extract.message, transform.message)
```
Comment (spheric) [inline spec/mable_etl/control_spec.rb:62]: move this stub up to the top before block

File: spec/mable_etl/control_spec.rb [hunk 9]
```diff
@@ -2,28 +2,81 @@
...
+          control.process
+          expect(dummy_logger.info).to eq(extract.message)
+          expect(dummy_logger.info).to eq(transform.message)
+        end
+
+        context 'when it is unsuccessful' do
+          let(:success) { false }
+          it 'it returns a logger error' do
+            dummy_logger = double
+            allow(dummy_logger).to receive(:error).and_return(extract.message, transform.message)
```
Comment (spheric) [inline spec/mable_etl/control_spec.rb:73]: move this stub up to the top before block

File: spec/mable_etl/control_spec.rb (hunk 8 above)
Comment (StaceyBros) [reply spec/mable_etl/control_spec.rb:62]: The stub only works in the block

File: spec/mable_etl/control_spec.rb (hunk 9 above)
Comment (StaceyBros) [reply spec/mable_etl/control_spec.rb:73]: The stub only works in the example. It does not work in the before block https://relishapp.com/rspec/rspec-mocks/v/3-12/docs/configuring-responses/returning-a-value#specify-different-return-values-for-multiple-calls

**Review by StaceyBros — APPROVED** [review]
LGTM

**Pull request merged**